sesman/chansrv/input_ibus.c

```diff
@@ -220,8 +220,21 @@ xrdp_input_unicode_init(void)
         return 0;
     }
 
-    /* Wait becasue ibus daemon may not be ready in first login. Do we have a flag to avoid busy waiting? */
-    sleep(3);
+    /* Wait because the ibus daemon may not be ready on first login */
+    const char *addr = ibus_get_address();
+    unsigned int cnt = 0;
+    while (!addr && cnt < 10)
+    {
+        usleep(500 * 1000); // half a second
```

Comment:
I think we can increase the timeout a little longer, such as 3-5 seconds. Longer timeout has no side effects because it will end as soon the iBus is ready. We can cover 99.99%+ computers for that.

Reply:
@seflerZ - I think the timeout is 5 seconds with the above (10 * 0.5 secs). Is that enough?